external/js/cothority/src/protobuf/marshaller.ts

```diff
@@ -0,0 +1,63 @@
+import Long from "long";
+
+/**
+ * IStorage represents a storage backend - either a local cache, or a db
+ * that stays around between sessions.
+ */
+export interface IStorage {
+    get(key: string): Promise<Buffer | undefined>;
+    set(key: string, value: Buffer): Promise<void>;
+}
+
+/**
+ * Marshaller takes care of Buffer and Long when a JSON.stringified object
+ * is JSON.parsed.
+ */
+export class Marshaller implements IStorage {
+    constructor(private storage: IStorage = new LocalCache()) {
+    }
+
+    async get(key: string): Promise<Buffer | undefined> {
+        return this.storage.get(key);
+    }
+
+    async set(key: string, value: Buffer): Promise<void> {
+        return this.storage.set(key, value);
+    }
+
+    async getObject<T>(key: string): Promise<T | undefined> {
+        const buf = await this.get(key);
+        if (buf === undefined) {
+            return undefined;
+        }
+        return JSON.parse(buf.toString(), (_, value) => {
+            if (value && typeof value === "object") {
+                if (value.type && value.type === "Buffer") {
```

Comment:
Is `value.type` always existing? I don't see it in the [nodejs/Buffer documentation](https://nodejs.org/api/buffer.html)
If so, why not use it for `Long` also? Else, better use `"type" in value`.

Or if it is wanted to be a falsy value, same idea as before.

Reply:
deleted code - that's what you get when you go too deep in a WIP PR...